Improve documentation (#409)
ESSS/alfasim-sdk: docs/source/plugins/06_solver_hooks.rst

Solver Hooks
============

The present section describes all solver `Hooks` available in |alfasim| plugin infrastructure.
Solver `Hooks` are |alfasim|'s pre-defined functions that allow plugin developer to modify or extend |alfasim|'s Solver.
As already informed in :ref:`Quick Start <quick-start-section>` section, once created a plugin using ``new`` option

> in |sdk|'s CLI a new file named ``plugin.c`` will be available to implement those `hooks`.

Reviewer: This seems to be not up to date. Currently, the file created is `plugin.cpp` and not `plugin.c`. Or am I missing something?

Author: You're right. Fixed.

The `hooks` described in this section are related to plugin secondary variables that were registered in the python config
file, as already explained in :ref:`solver_customization` section. They are called `secondary variables` because they

> are not obtained from |alfasim|'s Solver like the `pressure`, `temperature`, `volume fractions` and `velocities`, called primary variables istead.

Reviewer: ```suggestion
are not obtained from |alfasim|'s Solver like the `pressure`, `temperature`, `volume fractions` and `velocities`, called primary variables instead.
```

Author: Done.

Source Terms
------------

> The `hooks` showed in this section can be considered as the most important since they allow the plugin to change the

Reviewer: ```suggestion
The `hooks` showed in this section can be considered as the most important hooks since they allow the plugin to change the
```

Author: I think I'll keep it as it is because using `hooks` again sounds redundant.

----------------------

When a new phase is added to the hydrodynamic model using the :py:class:`~alfasim_sdk.AddPhase` type, it is possible

> to consider the deposition inside the pipeline walls. In this case, the thickness of the deposited layer on a given
> phase is calculated through the usage of :py:func:`~alfasim_sdk._internal.hook_specs.update_internal_deposition_layer`. By default, the layer

Reviewer: ```suggestion
to consider the deposition inside the pipeline walls. In this case, the thickness of the deposited layer on a given
phase should be calculated through the usage of :py:func:`~alfasim_sdk._internal.hook_specs.update_internal_deposition_layer`. By default, the layer
```

Proposing this change, because `is calculated` gives the idea that the deposition layer is calculated by alfasim instead of the user calculating it and imposing the value in the hook

Author: I agree! Done.

> .. note ::

Author: @ggrbill  I didn't get this note. Could you help me, pls?

Reviewer 2: This is tricky to explain.

I think we could change the name of things. Like the `Two-phase (Gas-Liquid) System of Mechanistic Model` Could be just `Two-Phase Mechanistic Model. 

Then, this note could be:
"
Note that the two-phase (Gas-Liquid) Mechanistic Model can be used for both two and three-phase flows (Fluid FLow Type selected by the user in ALFAsim). When no Liquid-Liquid System is implemented by the plugin, the three-phase flow is adapted to work with the traditional two-phase Mechanistic Model, in which all liquid phases are considered as only one.
"

See if you understand it. Perhaps it needs a little bit more improvement.

Author: I see that this is hard to explain in this context since it requires a deeper knowledge of the hydrodynamic modeling. But I think that your suggestion is much better than the original text. Thanks!
